www/js/lib/uiUtil.js

```diff
@@ -32,6 +32,48 @@ if (webpMachine) {
+
+    /**
+     * Displays a Bootstrap alert or confirm dialog box depending on the options provided
+     * 
+     * @param {String} message The alert message to display in the body of the modal 
+     * @param {String} label The modal's label or title which appears in the header (optional, Default = "Confirmation" or "Message")
+     * @param {Boolean} isConfirm If true, the modal will be a confirm dialog box, otherwise it will be a simple alert message 
+     * @param {String} declineConfirmLabel The text to display on the decline confirmation button (optional, Default = "Cancel") 
+     * @param {String} approveConfirmLabel  The text to display on the approve confirmation button (optional, Default = "Confirm")
+     * @param {String} closeMessageLabel  The text to display on the close alert message button (optional, Default = "Okay")
+     * @returns {Promise<Boolean>} A promise which resolves to true if the user clicked Confirm, false if the user clicked Cancel/Okay, backdrop or the cross(x) button
+     */
+    function systemAlert(message, label, isConfirm, declineConfirmLabel, approveConfirmLabel, closeMessageLabel) {
+        declineConfirmLabel = declineConfirmLabel || document.getElementById("declineConfirm").textContent;
+        approveConfirmLabel = approveConfirmLabel || document.getElementById("approveConfirm").textContent;
+        closeMessageLabel = closeMessageLabel || document.getElementById("closeMessage").textContent;
+        label = label || (isConfirm ? "Confirmation" : "Message");
+        return new Promise(function (resolve, reject) {
+            if (!message) reject("Missing body message");
+            // Set the text to the modal and it's buttons
+            document.getElementById("approveConfirm").textContent = approveConfirmLabel;
+            document.getElementById("declineConfirm").textContent = declineConfirmLabel;
+            document.getElementById("closeMessage").textContent = closeMessageLabel;
+            document.getElementById("modalLabel").textContent = label;
+            document.getElementById("modalText").textContent = message;
+            // Display buttons acc to the type of alert
+            document.getElementById("approveConfirm").style.display = isConfirm ? "block" : "none";
+            document.getElementById("declineConfirm").style.display = isConfirm ? "block" : "none";
+            document.getElementById("closeMessage").style.display = isConfirm ? "none" : "block";
+            // Display the modal
+            $("#alertModal").modal("show");
+            // When hide model is called, resolve promise with true if hidden using approve button, false otherwise
+            $("#alertModal").on("hide.bs.modal", function () {
+                const closeSource = document.activeElement;
+                if (closeSource.id === "approveConfirm") {
+                    resolve(true);
+                } else {
+                    resolve(false);
+                }
```

Comment:
Are you missing code here to reset the buttons to their original values? I think that to do that, you would need to define above, before line 48, something like `defaultValDeclineConfirm = document.getElementById("declineConfirm").textContent;`, and the same for the other values. Obviously then use this variable in lines 48, 49 and 50. Then here, you can reset the textContent of each to their default values, right?

Reply:
Yes, mossroy mentioned this earlier, missed it somehow. Looking into it.

Reply:
@Jaifroid  I think this won't work. Like, if a call is made to this function twice. First with a custom button label, say, No. And, the second with no custom label passed. Then in the second function run, the default label value, `defaultValDeclineConfirm` will pick up "No" from the DOM. So, this won't solve our problem ig. 

A solution can be to hardcode the default value rather than picking it up from the DOM, but this will again pose the initial problem of future UI translation as pointed out by @mossroy.

But, I think we are also setting the default values for label variable in the JS, isn't that a problem for UI translations? 
https://github.com/kiwix/kiwix-js/blob/9dc5281932f18ab493b7c81809ceace39ec12a0b/www/js/lib/uiUtil.js#L51

CC: @mossroy

Reply:
If we want to have a chance to merge this PR before the release, you should revert to the hard-coded labels you had before.
We'll deal with the translations later

Reply:
Done, please have a look.

Reply:
Actually, overnight I thought about this problem, and realized we need to set the default values as constants outside (just above) this function, so that they are invariable and will never be changed by running the function. In fact, it would be a coding mistake if this code block runs twice before defaults are restored, but humans make mistakes, so we shouldn't assume it will never happen. 😯

I agree we should leave this as hard coded now, and fix it when we tackle #13.

Reply:
> But, I think we are also setting the default values for label variable in the JS, isn't that a problem for UI translations?

It's a good point. I think it depends on the framework. A good one should allow us to have translations for strings in JS as well as in HTML. I think mossroy's point is also that such backend functions shouldn't have a dependency on the UI for good code architecture. See #813.